src/vdom.rs

```diff
@@ -288,9 +324,54 @@ impl<Ms, Mdl, ElC: View<Ms> + 'static, GMs: 'static> App<Ms, Mdl, ElC, GMs> {
         since = "0.4.2",
         note = "Please use `AppBuilder.build_and_start` instead"
     )]
-    pub fn run(self) -> Self {
+    pub fn run(mut self) -> Self {
+        let AppRunCfg {
+            mount_type,
+            into_after_mount,
+            ..
+        } = self.run_cfg.take().expect(
+            "run_cfg should be set in App::new which is called from AppBuilder::build_and_start",
+        );
+
         // Bootstrap the virtual DOM.
-        self.data.main_el_vdom.replace(Some(self.bootstrap_vdom()));
+        self.data
+            .main_el_vdom
+            .replace(Some(self.bootstrap_vdom(mount_type)));
+
+        // Can this simply be `self.update`?
+        let mut orders = OrdersContainer::new(self.clone());
+        let builder::AfterMount {
+            model,
+            url_handling,
+        } = into_after_mount.into_after_mount(routing::initial_url(), &mut orders);
+
+        self.data.model.replace(Some(model));
+
+        // TODO: Does this go before or after setting up the routes listener?
+        // TODO: Does this go before or after the initial render? The effects are created before
+        // TODO: the render, but after the orders. But old behavior let this run before the orders.
+        match url_handling {
+            UrlHandling::PassToRoutes => {
+                let url = routing::initial_url();
+                if let Some(routes) = self.data.routes.borrow().as_ref() {
+                    if let Some(routing_msg) = routes(url) {
+                        (self.cfg.update)(
+                            routing_msg,
+                            &mut self.data.model.borrow_mut().as_mut().unwrap(),
+                            &mut orders,
+                        );
+                    }
+                }
+            }
+            UrlHandling::None => (),
+        };
```

Comment:
I think we can move it under the comments lines:
```
// Update the state on page load, based
// on the starting URL. Must be set up on the server as well.
```
What do you think?

Reply:
If we match old behavior, this needs to happen before the window listeners are set up. If that's not necessary behavior to keep (or even behavior we shouldn't keep), then yes.

Reply:
Don't look at the old code / behavior very much, there is a chance it was wrong.

Write it how it makes sense to you and it would be nice to add comment why you've chosen that position/order so we don't need to think about it in the future.

And it would be also nice to test it with an example, but the only one with routing is TodoMVC and it looks buggy / incomplete according to its code. 
So we should write some more examples with routing and fix TodoMVC in the future.
 (cc @David-OConnor )

Reply:
I think I'm going to change this. Right now, I think this function is fairly confusing. What we have here is as follows:

`IntoAfterMount` generates a series or `Effect`s.
`model` is set.
`routing` function is called and creates a message.
`update` is called on the routing message. (This message is created after the effects in init!)
Previously generated effects are called. (These messages are created before the messages in `IntoAfterMount`!)

I propose that we simply toss the routing message onto the queue and process them in the order that they are created.

Reply:
@AlterionX 
I think you have the deepest knowledge of this problem / know the best solution because you are refactoring it for a while. And it makes sense to me and I trust you with it - so I agree with your suggestions. And it should be tested with the real projects - it will be, once it's in `master`.